File: app/src/main/java/com/app/sid/funwithflags/view/base/BaseActivity.java
package com.app.sid.funwithflags.view.base;

import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;

/**
 * @author Sidhanshu
 * This activtiy is extented by every activity created in the application, used for better
 * handeling generic metods and perform operation which could be common throughtout the application.
 */
public abstract class BaseActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Override
    protected void onSaveInstanceState(Bundle outState) {
        super.onSaveInstanceState(outState);

    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
    }

}

File: app/src/main/java/com/app/sid/funwithflags/view/BaseActivity.kt
package com.app.sid.funwithflags.view

import android.os.Bundle
import androidx.annotation.LayoutRes
import dagger.android.support.DaggerAppCompatActivity

abstract class BaseActivity(@LayoutRes private val layoutResId: Int) : DaggerAppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(layoutResId)
    }
}

